src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -106,7 +108,11 @@ public Future<DeploymentResult> mergeInNewConfig(Deployment deployment,
         } else {
             logger.atInfo().log("Deployment is configured to skip update policy check,"
                     + " not waiting for disruptable time to update");
-            updateActionForDeployment(newConfig, deployment, activator, totallyCompleteFuture);
+            // use executor service to execute updateActionForDeployment
+            // prevents default deployment cancellation from directly interrupting
+            executorService.execute(() -> updateActionForDeployment(newConfig, deployment, activator,
```

Comment:
Do we need to wait for this future to complete?

Reply:
we do not need to wait for this future to complete, instead we wait on `CompletableFuture totallyCompleteFuture` that's being returned